modules/reindex/src/javaRestTest/java/org/elasticsearch/index/reindex/remote/ReindexFromOldRemoteIT.java

```diff
@@ -92,8 +93,16 @@ private void oldEsTestCase(String portPropertyName, String requestsPerSecond) th
                 for (int i = 0; i < DOCS; i++) {
                     assertThat(result, containsString("\"_id\" : \"testdoc" + i + "\""));
                 }
-            } finally {
                 oldEs.performRequest(new Request("DELETE", "/test"));
+                indexDeleted = true;
+            } finally {
+                if (indexDeleted == false) {
+                    try {
+                        oldEs.performRequest(new Request("DELETE", "/test"));
+                    } catch (Exception deleteException) {
+                        logger.warn("Exception deleting index", deleteException);
+                    }
+                }
```

Comment:
I do not think it is required to delete the index again in case of exception.
It is enough to catch and log it inside the finally block to propagate the original exception.

Reply:
Notice that with the way it is structured now, it could have failed earlier too (for instance in one of the assertions) and in those cases the delete need to be done.

I suppose we could move the line [`idexDeleted = true`](https://github.com/elastic/elasticsearch/pull/85425/files#diff-5cead9923abe48a90807d6356afc5b0805910bd4157cae1b129624fe3461171dR97) before the attempt to delete the index if we want to avoid the double delete attempt, that seems fine to me.

Reply:
I think the only delete attempt need to happen in `finally` block (wrapped into its own `try`).

```
try {
 createIndex;
 createTestDocuments;
 assert;
} finally {
  try { delete; }
}
```

This way it will not swallow original exception (from assert) in case index deletion fails.

Reply:
I mean I would prefer a failed delete in a success case not disappear to a warn message since that is still a concern. But I made a change in this area.